slowlog_functions.php

```diff
@@ -0,0 +1,1439 @@
+						}
+					} elseif ($in_into) {
+						if (is_reserved_word($t) && $t != 'table') {
+							slowlog_debug("RES WORD: " . $t);
+							$table_mod = true;
+						} elseif ($bparen) {
+							$table_mod = true;
+						} elseif ($t == 'table') {
+							$table_mod = false;
+						} elseif (!$table_mod) {
+							if ($t != '' && !is_reserved_word($t)) {
+								$table = parseTable($t);
+								$tables[$table] = $table;
+								slowlog_debug('INTO: ' . $table);
+								$table_mod = true;
+							} elseif (is_reserved_word($t)) {
+								$table_mod = false;
+							} else {
+								$table_mod = true;
+							}
+						}
+					} elseif ($in_join) {
+						if ($t != '' && !is_reserved_word($t) && !$table_mod) {
+							$table = parseTable($t);
+							$tables[$table] = $table;
+							slowlog_debug('JOIN: ' . $table);
+							$table_mod = true;
+						}
+					} elseif ($in_update) {
+						if ($t == 'force' || $t == 'index') {
+							$table_mod = true;
+						} elseif ($t == 'inner' || $t == 'left' || $t == 'right' || $t == 'outer') {
+							$table_mod = false;
+						} elseif (!$table_mod) {
+							if ($t != '' && !is_reserved_word($t)) {
+								$table = parseTable($t);
+								$tables[$table] = $table;
+								slowlog_debug('UPDATE: ' . $table);
+							} elseif (is_reserved_word($t)) {
+								$table_mod = false;
+							} else {
+								$table_mod = true;
+							}
+						}
+					} elseif ($in_truncate) {
+						$table_mod = false;
+						if (is_reserved_word($t)) {
+							$table_mod = false;
+						} elseif ($t != '') {
+							$table = parseTable($t);
+							$tables[$table] = $table;
+							slowlog_debug('TRUNCATE: ' . $t);
+						} else {
+							$table_mod = true;
+						}
+					} elseif ($in_table) {
+						if (is_reserved_word($t)) {
+							$table_mod = false;
+						} elseif ($t != '' && !$table_mod) {
+							$table = parseTable($t);
+							$tables[$table] = $table;
+							slowlog_debug('TABLE: ' . $t);
+
+							$table_mod = true;
+						}
+
+					} else {
+						slowlog_debug('Something: ' . $t);
+					}
+			}
+		}
+
+		if (cacti_sizeof($tables)) {
+			foreach($tables as $t) {
+				$sql[] = '(' . $logid . ', ' . $row['logentry'] . ', ' . db_qstr($t) . ')';
+			}
+		} else {
+			slowlog_debug("No tables found!!!: $query");
+		}
+	}
+
+	if (cacti_sizeof($sql)) {
+		cacti_log('Post Processing Tables associated: ' . cacti_sizeof($sql), false, 'SLOWLOG');
+
+		$sqls = array_chunk($sql, 500);
+
+		foreach($sqls as $sql) {
+			db_execute($sql_prefix . implode(', ', $sql) . $sql_suffix);
+		}
+	}
+}
+
+function parseTable($table) {
+	$table = trim($table, ';)');
+	$table = str_replace('`', '', $table);
+	$parts = explode('.', $table);
+
+	// Prune the schema
+	if (cacti_sizeof($parts) > 1) {
+		$table = $parts[1];
+	}
+
+	// Remove beginning braces
+	$parts = explode('(', $table);
+	if (cacti_sizeof($parts) == 1) {
+		return trim($table, '\'`');
+	} else {
+		return trim($parts[0], '\'`');
+	}
+}
```

Comment:
You can reduce this to > 1 and then just use the $parts[0].  The only time it would not be populated is if the table variable was already null/false